src/sentry/static/sentry/app/components/alertMessage.jsx

```diff
@@ -5,23 +5,28 @@ import {t} from '../locale';
 
 const AlertMessage = React.createClass({
   propTypes: {
-    className: React.PropTypes.string,
-    id: React.PropTypes.number.isRequired,
-    message: React.PropTypes.string.isRequired,
-    type: React.PropTypes.string,
-    url: React.PropTypes.string
+    alert: React.PropTypes.shape({
+      id: React.PropTypes.string,
+      message: React.PropTypes.string.isRequired,
+      type: React.PropTypes.oneOf([
+        'success',
+        'error',
+        'warning'
+      ]),
+      url: React.PropTypes.string
+    })
   },
 
   mixins: [PureRenderMixin],
 
   closeAlert: function() {
-    AlertActions.closeAlert(this.props.id);
+    AlertActions.closeAlert(this.props.alert);
   },
 
   render: function() {
-    let className = this.props.className || 'alert';
-    if (this.props.type !== '') {
-      className += ' alert-' + this.props.type;
+    let className = 'alert';
+    if (this.props.alert.type !== '') {
```

Comment:
I'm not sure if this was correct before — won't this potentially result in `alert-undefined`?